vm.go

```diff
@@ -225,6 +263,53 @@ func (vm *VM) EvaluateSnippetMulti(filename string, snippet string) (files map[s
 	return
 }
 
+// FindDependencies returns a sorted array of unique transitive dependencies (via import or importstr)
+// from all the given `importedPaths` which are themselves excluded from the returned array.
+// The `importedPaths` are parsed as if they were imported from a Jsonnet file located at `importedFrom`.
+func (vm *VM) FindDependencies(importedFrom string, importedPaths []string) ([]string, error) {
+	var nodes []*ast.Node
+	var stackTrace []traceFrame
+	deps := make(map[string]struct{})
+
+	for i, filePath := range importedPaths {
+		node, foundAt, err := vm.ImportAST(importedFrom, filePath)
+		if err != nil {
+			return nil, err
+		}
+
+		// Overwrite `importedPaths` with their canonical paths.
+		importedPaths[i] = foundAt
```

Comment:
The slices are mutable – this changes the argument that was passed to us. It can be very confusing for the user of this library. Create a new slice instead.

Reply:
Got it.